src/numbers.rs

```diff
@@ -77,41 +80,48 @@ pub fn float_parse_bytes(s: &[u8]) -> IntFloat {
             b'0'..=b'9' => {
                 int_part = match int_part.checked_mul(10) {
                     Some(i) => i,
-                    None => return IntFloat::Err,
+                    None => return (IntFloat::Err, None),
                 };
                 int_part = match int_part.checked_add((digit & 0x0f) as i64) {
                     Some(i) => i,
-                    None => return IntFloat::Err,
+                    None => return (IntFloat::Err, None),
                 };
             }
             b'.' => {
                 found_dot = true;
                 break;
             }
-            _ => return IntFloat::Err,
+            _ => return (IntFloat::Err, None),
         }
     }
 
     if found_dot {
         let mut result = int_part as f64;
-        let mut div = 10_f64;
+        let mut decimal_integers: u64 = 0;
+        let mut decimal_digits: usize = 0;
+
         for digit in bytes {
             match digit {
                 b'0'..=b'9' => {
-                    result += (digit & 0x0f) as f64 / div;
-                    div *= 10_f64;
+                    decimal_digits += 1;
+                    decimal_integers = decimal_integers * 10 + (digit & 0x0f) as u64;
```

Comment:
unless this resolves to the exact same logic we had before (and I don't think it does), I would prefer not to change it. As we discovered in jiter, parsing floats properly is an extremely complex problem, that requires at least 4 different algorithms for different cases.

We should either leave this as-is, so at least it's behavior doesn't change, or replace this logic with [`lexical-parse-float`](https://crates.io/crates/lexical-parse-float).

Reply:
Agreed, reverted and addressed here: https://github.com/pydantic/speedate/issues/53